### PR TITLE
[SB] Remove msrest dependency

### LINKED ISSUE #26571: [EventHubs&ServiceBus] remove msrest import from _amqp_message
@kashifkhan pointed out that we're using msrest [[here](https://github.com/Azure/azure-sdk-for-python/blob/73d1c4dd2f6dd4f7b825db998542e374fe6b4c0e/sdk/servicebus/azure-servicebus/azure/servicebus/amqp/_amqp_message.py#L14)] and this should be removed.

### PR DESCRIPTION
**Pull request by kashifkhan**
#26571 

Replaced msrest with the azcore equivalent, we should not need msrest as part of setup.py either for SB.

### CODE REVIEW

**Review by swathipil — APPROVED** [review]
LGTM!

File: sdk/servicebus/azure-servicebus/setup.py [hunk 1]
```diff
@@ -65,7 +65,6 @@
     install_requires=[
         "uamqp>=1.5.1,<2.0.0",
         'azure-common~=1.1',
-        'msrest>=0.6.17,<2.0.0',
```
[the remark this answers is not in the document]
Comment (kashifkhan) [reply sdk/servicebus/azure-servicebus/setup.py:68]: only in mgmt

[the remark this answers is not in the document]
Comment (swathipil) [reply sdk/servicebus/azure-servicebus/setup.py:68]: ah, in that case, would we have to keep this in until we regenerate mgmt?

[the remark this answers is not in the document]
Comment (kashifkhan) [reply sdk/servicebus/azure-servicebus/setup.py:68]: good catch

**Pull request closed without merging**